packages/fxa-event-broker/lib/notificationProcessor.ts

```diff
@@ -130,85 +87,92 @@ class ServiceNotificationProcessor {
     this.webhookService.stop();
   }
 
+  private async handleDeleteEvent(message: deleteSchema) {
+    const clientIds = await this.db.fetchClientIds(message.uid);
+    clientIds.forEach(async clientId => {
+      const topicName = this.topicPrefix + clientId;
+      const messageId = await this.pubsub
+        .topic(topicName)
+        .publishJSON({ event: message.event, uid: message.uid });
+      this.logger.debug('publishedMessage', { topicName, messageId });
+    });
+  }
+
+  private async handleLoginEvent(message: loginSchema) {
+    // Sync and some logins don't emit a clientId, so we have nothing to track
+    if (!message.clientId) {
+      this.logger.debug('unwantedMessage', {
+        message
+      });
+      return;
+    }
+    await this.db.storeLogin(message.uid, message.clientId);
+  }
+
+  private async handleSubscriptionEvent(message: subscriptionUpdateSchema) {
+    const clientIds = await this.db.fetchClientIds(message.uid);
+
+    // Split the product capabilities by clientId each capability goes to
+    const notifyClientIds: { [clientId: string]: string[] } = {};
+    message.productCapabilities
+      .map(item => item.split(':'))
+      .forEach(([clientId, capability]) => {
+        if (notifyClientIds[clientId]) {
+          notifyClientIds[clientId].push(capability);
+        } else {
+          notifyClientIds[clientId] = [capability];
+        }
+      });
+
+    const baseMessage = {
+      capabilities: [],
+      event: message.event,
+      isActive: message.isActive,
+      uid: message.uid
+    };
+
+    Object.entries(notifyClientIds)
+      .filter(([clientId]) => clientIds.includes(clientId))
+      .forEach(async ([clientId, capabilities]) => {
+        const topicName = this.topicPrefix + clientId;
+        const rpMessage = Object.assign(
+          {},
+          {
+            ...baseMessage,
+            capabilities
+          }
+        );
+
+        // TODO: Failures to publish due to missing queue should be Sentry reported.
```

Comment:
This refactoring might be a good moment to file a followup issue and add the issue number to this TODO.

Reply:
Yup, a new issue was opened today to add Sentry reporting.